File: big-o/02-js/nemo2.js
const nemo = ['nemo'];
const everyone = ['dory', 'burce', 'marlin', 'nemo', 'gill', 'bloat', 'nigel', 'squirt', 'darla', 'hank'];
const large = new Array(100_000).fill('nemo');

const findNemo = array => {
  for (let i = 0; i < array.length; ++i){
    if (array[i] == 'nemo') {
      console.log('Found NEMO!');
    }
  }
}

findNemo(large); // O(n) -> Linear Time

const logFirstTwoBoxes = boxes => {
  console.log(boxes[0]);
  console.log(boxes[1]);
}

logFirstTwoBoxes(everyone); // Constant Time 

File: big-o/02-rs/nemo2.rs
use std::time::Instant;

fn find_nemo(array: &[&str]) {
    let now = Instant::now();

    for item in array {
        if *item == "nemo" {
            println!("Found NEMO!");
        }
    }

    let elapsed = now.elapsed();
    println!(
        "find_nemo function call took {} milliseconds",
        elapsed.as_millis()
    );
}

fn main() {
    let nemo = ["nemo"];
    let everyone = ["dory", "bruce", "marlin", "nemo", "gill", "bloat", "nigel", "squirt", "darla", "hank"];
    let large = vec!["nemo"; 100_000];
    find_nemo(&large);
} 
 
